agent.go

```diff
@@ -468,8 +468,12 @@ func (n *network) addDriverWatches() {
 
 	c := n.getController()
 	for _, tableName := range n.driverTables {
-		ch, cancel := c.agent.networkDB.Watch(tableName, n.ID(), "")
 		c.Lock()
+		if c.agent == nil {
```

Comment:
`c.agent` cannot be made nil if we hold the lock.
This change adds some coordination, so that when we access c.agent and we assert it is not nil, it we can modify its members assured it cannot be made nil in the meanwhile.